subsys/bluetooth/audio/ascs.c

```diff
@@ -106,6 +106,11 @@ NET_BUF_SIMPLE_DEFINE_STATIC(ase_buf, ASE_BUF_SIZE);
 static int control_point_notify(struct bt_conn *conn, const void *data, uint16_t len);
 static int ascs_ep_get_status(struct bt_bap_ep *ep, struct net_buf_simple *buf);
 
+static bool check_cis_link_loss(struct bt_ascs_ase *ase)
+{
+	return ase->ep.reason == BT_HCI_ERR_CONN_TIMEOUT;
```

Comment:
Link loss happens when the remote part stops sending PDUs for a specific time (supervision timeout). The error codes you've mentioned mean:
`BT_HCI_ERR_REMOTE_USER_TERM_CONN`: The remote user terminated connection. Meaning the connection was explicitly disconnected e.g. HCI Disconnect commands was issued on remote part.
`BT_HCI_ERR_TERM_DUE_TO_MIC_FAIL`: The message integrity check failed meaning the encrypted PDU was corrupted.

There was a discussion on Jira: https://bluetooth.atlassian.net/jira/software/c/projects/ES/issues/ES-22692. I've added a comment there as I see no solution provided. There are 3 links linked but not accessible for me.

Reply:
I've had offline discussion with @sjanc, and we agreed that the ASE transition to QoS shall be for each CIS disconnection not preceded by ASE Release operation. I'll update this PR, so that the related PTS tests will pass

Reply:
`BT_HCI_ERR_CONN_FAIL_TO_ESTAB` may be a special case though, as that will come if the CIS was not established in the first place, but it will still trigger a call to disconnected. If we are in the enabling state and we get `BT_HCI_ERR_CONN_FAIL_TO_ESTAB`, we should, IMO, stay in the enabling state.

Reply:
As per spec, the autonomous transition shall happen if ASE is in Disabling or Streaming state only